www/js/modules/charts/controllers/HistogramChartController.js

```diff
@@ -199,19 +199,25 @@ export default class HistogramChartController {
 
         let that = this;
 
+		// must remove histogram to update histogram
+	    d3.select('#myHistogram123').remove();
+		this.$histogram = null;
         if (!this.$histogram) {
 
             this.$histogram = this.$svg
                 .append('g')
-                .attr('class', 'histogram');
+                .attr('class', 'histogram')
+				.attr('id', 'myHistogram123');
 
         }
 
-        this.$histogram
-            .attr('transform', `translate(${this.histogramChartModel.svgPaddingX}, ${this.histogramChartModel.svgPaddingY})`)
+		let bars = this.$histogram
+            .attr('transform', 'translate(' + [this.histogramChartModel.svgPaddingX, this.histogramChartModel.svgPaddingY] + ')')
             .selectAll('rect')
-            .data(this.histogramChartModel.data)
-            .enter()
+            .data(this.histogramChartModel.data);
+
+		// have to separate out the enter() so that the update can happen otherwise it won't work correctly.
+        bars.enter()
             .append('rect')
```

Comment:
@tniswong one of the patterns of d3 is to attach the data with all the "static" attributes etc for the chart.  Then call the enter() function so that all the dynamic parts of the chart get updated.  However, I don't think we have to do this in our case since we are removing the histogram from the DOM entirely, but just wanted to be consistent.